src/lib/ratings/streakRatingSystem.ts

```diff
@@ -0,0 +1,128 @@
+import { isDefined } from "../utils";
+import {
+  type MatchWithRatings,
+  type PlayerWithRating,
+  type RatingSystem,
+} from "./rating";
+
+export interface StreakRating {
+  points: number;
+  currentWinStreak: number;
+  currentLoseStreak: number;
+}
+
+export interface StreakConfig {
+  basePoints: number;
+  streakMultiplier: number;
+  maxStreakMultiplier: number;
+  streakBonus: number;
+}
+
+export function streak(config?: StreakConfig): RatingSystem<StreakRating> {
+  const selectedConfig: StreakConfig = config ?? {
+    basePoints: 25, // Base points awarded for a win
+    streakMultiplier: 1.2, // Multiplier increases by this factor per win
+    maxStreakMultiplier: 15, // Maximum multiplier cap (roughly 15 win streak)
+    streakBonus: 10, // Here to reward players early in their streaks, the multiplier overshadow this on longer streaks
+  };
+
+  function calculateWinPoints(streak: number): number {
+    return calculatePoints(
+      streak,
+      selectedConfig.streakMultiplier,
+      selectedConfig.maxStreakMultiplier,
+      selectedConfig.streakBonus,
+      selectedConfig.basePoints,
+    );
+  }
+
+  function calculateLosePoints(loseStreak: number): number {
+    return -calculatePoints(
+      loseStreak,
+      selectedConfig.streakMultiplier,
+      selectedConfig.maxStreakMultiplier,
+      selectedConfig.streakBonus,
+      selectedConfig.basePoints,
+    );
+  }
+
+  function calculatePoints(
+    streak: number,
+    streakMultiplier: number,
+    maxStreakMultiplier: number,
+    streakBonus: number,
+    basePoints: number,
+  ): number {
+    const multiplier = Math.min(
+      maxStreakMultiplier,
+      Math.pow(streakMultiplier, streak > 1 ? streak : 0),
+    );
+    return Math.round(basePoints * multiplier + (streak > 1 ? streakBonus : 0));
+  }
```

Comment:
Simple because of fewer lines, IMO, low LOGAF

```suggestion
  function calculateLosePoints(loseStreak: number): number {
    return -calculateWinPoints(loseStreak);
  }

  function calculateWinPoints(
    winStreak: number,
  ): number {
    const {
      streakMultiplier,
      maxStreakMultiplier,
      streakBonus,
      basePoints,
    } = selectedConfig;
    const multiplier = Math.min(
      maxStreakMultiplier,
      Math.pow(streakMultiplier, winStreak > 1 ? winStreak : 0),
    );
    return Math.round(basePoints * multiplier + (winStreak > 1 ? streakBonus : 0));
  }
```

Reply:
Du har fået en mellemvej, så der er lidt mindre linjer.